From 735e2fb020f35c30b3b0da428329c2e52e1e36aa Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Sun, 15 Apr 2018 20:56:15 +0200
Subject: [PATCH 1/2] selinux: add domain for snap

Snap needs to read vendor files on treble enabled devices

Change-Id: Ia4f3da3fa20ff894b5b202f6691e8f2c03f06e7f
---
 common/private/seapp_contexts |  1 +
 common/private/snap_app.te    | 10 ++++++++++
 2 files changed, 11 insertions(+)
 create mode 100644 common/private/seapp_contexts
 create mode 100644 common/private/snap_app.te

diff --git a/common/private/seapp_contexts b/common/private/seapp_contexts
new file mode 100644
index 0000000..a9e2d48
--- /dev/null
+++ b/common/private/seapp_contexts
@@ -0,0 +1 @@
+user=_app isPrivApp=true seinfo=platform name=org.lineageos.snap domain=snap_app type=app_data_file
diff --git a/common/private/snap_app.te b/common/private/snap_app.te
new file mode 100644
index 0000000..64a46ab
--- /dev/null
+++ b/common/private/snap_app.te
@@ -0,0 +1,10 @@
+type snap_app, domain, coredomain;
+
+app_domain(snap_app)
+net_domain(snap_app)
+
+# Allow to read and execute libjni_*.so
+allow snap_app vendor_file:file rx_file_perms;
+
+# Execute libraries from RenderScript cache
+allow snap_app app_data_file:file { rx_file_perms };
-- 
2.17.1

